pkg/graveler/graveler.go

```diff
@@ -1451,47 +1451,42 @@ func (g *KVGraveler) Delete(ctx context.Context, repository *RepositoryRecord, b
 		return ErrWriteToProtectedBranch
 	}
 
-	deleteEntry := func(branch *Branch) error {
-		commit, err := g.RefManager.GetCommit(ctx, repository, branch.CommitID)
-		if err != nil {
-			return err
-		}
-
-		// check key in committed - do we need tombstone?
-		foundInCommitted := false
-		_, err = g.CommittedManager.Get(ctx, repository.StorageNamespace, commit.MetaRangeID, key)
-		if err != nil {
-			if !errors.Is(err, ErrNotFound) {
-				// unknown error
-				return fmt.Errorf("reading from committed: %w", err)
+	return g.safeBranchWrite(ctx, g.log.WithField("key", key).WithField("operation", "delete"),
+		repository, branchID, func(branch *Branch) error {
+			// check staging for entry or tombstone
+			val, err := g.getFromStagingArea(ctx, branch, key)
+			switch {
+			case err != nil:
+				if !errors.Is(err, ErrNotFound) {
+					// unknown error
+					return fmt.Errorf("reading from staging: %w", err)
+				}
+				// entry not found in staging - continue to committed check
+			case val == nil:
+				// found tombstone in staging, return ErrNotFound
+				return ErrNotFound
+			default:
```

Comment:
Consider doing it explicit for readability:
```suggestion
			case val != nil:
```

Reply:
prefer to catch all at this point